### PR TITLE
Also include NEQ in filters

### PR DESCRIPTION
**Pull request by otoolep**
This fixes != for field value comparisons.

### CODE REVIEW

File: cmd/influxd/server_integration_test.go [hunk 1]
```diff
@@ -588,6 +588,21 @@ func runTestsData(t *testing.T, testName string, nodes Cluster, database, retent
 			expected: `{"results":[{"series":[{"name":"cpu","tags":{"region":"us-east"},"columns":["time","mean"],"values":[["1970-01-01T00:00:00Z",15]]},{"name":"cpu","tags":{"region":"us-west"},"columns":["time","mean"],"values":[["1970-01-01T00:00:00Z",30]]}]}]}`,
 		},
 
+		// WHERE tag queries
+		{
+			reset: true,
```
Comment (otoolep) [inline cmd/influxd/server_integration_test.go]: Added some `=` unit tests for tags. These always worked, but I wanted some tests.

File: cmd/influxd/server_integration_test.go [hunk 2]
```diff
@@ -646,6 +661,10 @@ func runTestsData(t *testing.T, testName string, nodes Cluster, database, retent
 			expected: `{"results":[{"series":[{"name":"cpu","columns":["time","load"]}]}]}`,
 		},
 		{
+			query:    `select load from "%DB%"."%RP%".cpu where load != 100`,
+			expected: `{"results":[{"series":[{"name":"cpu","columns":["time","load"],"values":[["2009-11-10T23:01:02Z",80]]}]}]}`,
+		},
+		{
```
Comment (otoolep) [inline cmd/influxd/server_integration_test.go]: New test case that failed before the change to `database.go`.

File: influxql/parser_test.go [hunk 3]
```diff
@@ -221,6 +221,19 @@ func TestParser_ParseStatement(t *testing.T) {
...
+				IsRawQuery: true,
+				Fields:     []*influxql.Field{{Expr: &influxql.Wildcard{}}},
+				Sources:    []influxql.Source{&influxql.Measurement{Name: "cpu"}},
+				Condition: &influxql.BinaryExpr{
+					Op:  influxql.NEQ,
+					LHS: &influxql.VarRef{Val: "load"},
+					RHS: &influxql.NumberLiteral{Val: 100},
+				},
+			},
+		},
```
Comment (otoolep) [inline influxql/parser_test.go]: Just to make sure it wasn't a parsing problem.

**Comment by otoolep** [comment]
@pauldix @dgnorton

File: database.go [hunk 4]
```diff
@@ -405,14 +405,14 @@ func (m *Measurement) walkWhereForSeriesIds(expr influxql.Expr, filters map[uint
 	switch n := expr.(type) {
 	case *influxql.BinaryExpr:
 		// if it's EQ then it's either a field expression or against a tag. we can return this
-		if n.Op == influxql.EQ || n.Op == influxql.LT || n.Op == influxql.LTE || n.Op == influxql.GT ||
-			n.Op == influxql.GTE || n.Op == influxql.EQREGEX || n.Op == influxql.NEQREGEX {
+		switch n.Op {
+		case influxql.EQ, influxql.LT, influxql.LTE, influxql.GT, influxql.GTE, influxql.EQREGEX, influxql.NEQREGEX, influxql.NEQ:
```
Comment (otoolep) [inline database.go]: The fix is here -- add influxql.NEQ. I also changed to the code to use `switch` -- I think it's easier to read now.

File: database.go [hunk 5]
```diff
@@ -475,6 +475,8 @@ func (m *Measurement) walkWhereForSeriesIds(expr influxql.Expr, filters map[uint
 
 			// finally return the ids and say that we should include them
 			return ids, true, nil
+		default:
```
Comment (otoolep) [inline database.go]: This may not be the right thing to do, but it would have flagged the bug immediately.

**Comment by pauldix** [comment]
LGTM, :ship:. Since you were adding test cases for queries against tags, it wouldn't hurt to add some that test against multiple tags using `AND` and `OR` operators.

**Comment by otoolep** [comment]
Let me get back to that -- I'm working on tags in https://github.com/influxdb/influxdb/pull/2105, and will add more tests there.

**Pull request merged**